### PR TITLE
try that for clues

### PR DESCRIPTION
**Pull request by monsieurleberre**
### New Pull Request Submissions:

1. [ ] Have you followed the guidelines in our [Contributing document](https://github.com/atlascity/Community/tree/master/CONTRIBUTING.md)?
2. [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
3. [ ] I have added tests to cover my changes.
4. [ ] All new and existing tests passed.
5. [ ] Have you lint your code locally prior to submission?
6. [ ] Does your code follows the code style of this project?
7. [ ] Does your change require a change to the documentation.
    - [ ] I have updated the documentation accordingly.
9. [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
10. [ ] Have you inserted a keyword and link to the issues the PR closes in its descriptions (ex `closes #1`) ?
11. [ ] Is you branch up to date, have you integrated all the latest changes from develop and resolved conflicts ?

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: src/Catalyst.TestUtils/FileSystemBasedTest.cs [hunk 1]
```diff
@@ -123,8 +125,17 @@ protected virtual void Dispose(bool disposing)
             var regex = new Regex(CurrentTestName + @"_(?<timestamp>[\d]{16})");
 
             var oldDirectories = _testDirectory.Parent.EnumerateDirectories()
-               .Where(d => regex.IsMatch(d.Name)
-                 && string.CompareOrdinal(d.Name, _testDirectory.Name) == -1)
+               .Where(d =>
+                {
+                    var matches = regex.Matches(d.Name);
+                    if (matches.Count == 0) return false;
+                    if (!DateTime.TryParseExact(matches[0].Groups["timestamp"].Value,
```
Comment (nshCore) [inline src/Catalyst.TestUtils/FileSystemBasedTest.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4375968)

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/Catalyst.TestUtils/FileSystemBasedTest.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4375968&bid=14850814)

File: src/Catalyst.TestUtils/FileSystemBasedTest.cs [hunk 2]
```diff
@@ -123,8 +125,17 @@ protected virtual void Dispose(bool disposing)
             var regex = new Regex(CurrentTestName + @"_(?<timestamp>[\d]{16})");
 
             var oldDirectories = _testDirectory.Parent.EnumerateDirectories()
-               .Where(d => regex.IsMatch(d.Name)
-                 && string.CompareOrdinal(d.Name, _testDirectory.Name) == -1)
+               .Where(d =>
+                {
+                    var matches = regex.Matches(d.Name);
+                    if (matches.Count == 0) return false;
```
Comment (nshCore) [inline src/Catalyst.TestUtils/FileSystemBasedTest.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4375968)

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Catalyst.TestUtils/FileSystemBasedTest.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4375968&bid=14850814)

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- src/Catalyst.TestUtils/FileSystemBasedTest.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4375968&bid=14850814)

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- src/Catalyst.TestUtils/FileSystemBasedTest.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4375968&bid=14850814)

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- src/Catalyst.TestUtils/FileSystemBasedTest.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4375968&bid=14850814)

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- src/Catalyst.TestUtils/FileSystemBasedTest.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4375968&bid=14850814)

**Comment by monsieurleberre** [comment]
Green

**Pull request merged**